calendar/modules/listener/http_service.bal

```diff
@@ -79,30 +77,32 @@ service class HttpService {
                 calendar:Time? end = event?.end;
                 if (created is string && updated is string && 'start is calendar:Time && end is calendar:Time) {
                     if (created.substring(0, 19) == updated.substring(0, 19)) {
-                       if (self.isOnNewEventAvailable) {
-                           check callOnNewEventMethod(self.httpService, event);
-                       }
-                    } else  {
-                        if (self.isOnEventUpdateAvailable) {
-                            check callOnEventUpdateMethod(self.httpService, event);
+                        if (self.isOnNewEventAvailable) {
+                            check callOnNewEventMethod(self.httpService, event);
                         }
+                    } else if (self.isOnEventUpdateAvailable) {
+                        check callOnEventUpdateMethod(self.httpService, event);
                     }
-                } else {
-                    if (self.isOnEventDeleteAvailable) {
-                        check callOnEventDeleteMethod(self.httpService, event);
-                    }
+                } else if (self.isOnEventDeleteAvailable) {
+                    check callOnEventDeleteMethod(self.httpService, event);
                 }
+                check caller->respond(res);
             }
+        } else {
+            return error(INVALID_ID_ERROR);
```

Comment:
@chamil321 Could you please review this. If we use `return error`, will http module respond as error to the caller?

Reply:
If you return an error, it will be a 500 Internal Server Error response to the caller. The payload will be the error message.